### PR TITLE
:seedling: using kube api errors for work clients

### PR DESCRIPTION
**Pull request by skeeey**
## Related issue(s)

Fixes #

### CODE REVIEW

**Review by qiujian16 — COMMENTED** [review]
/approve

/assign @morvencao

File: pkg/cloudevents/work/source/client/manifestwork.go [hunk 1]
```diff
@@ -170,24 +185,32 @@ func (c *ManifestWorkSourceClient) List(ctx context.Context, opts metav1.ListOpt
...
+		return nil, errors.NewInternalError(err)
+	}
+	if !exists {
+		return nil, errors.NewNotFound(common.ManifestWorkGR, name)
 	}
 
 	patchedWork, err := utils.Patch(pt, lastWork, data)
 	if err != nil {
-		return nil, err
+		return nil, errors.NewInternalError(err)
```
Comment (morvencao) [inline pkg/cloudevents/work/source/client/manifestwork.go:213]: Nit: Could we make the patch error messages more fine-grained? For example, specify if the error is due to field conflicts or other issues.

Comment (skeeey) [reply pkg/cloudevents/work/source/client/manifestwork.go:213]: for conflicts, it seems hard to judge on the client part, but we can change this error with BadRequest, I think this more reasonable

**Review by morvencao — COMMENTED** [review]
/lgtm

File: pkg/cloudevents/work/source/client/manifestwork.go (hunk 1 above)
Comment (morvencao) [reply pkg/cloudevents/work/source/client/manifestwork.go:213]: ok, it makes sense.

**Pull request merged**